### PR TITLE
Add chunkForDelete() method to Query Builder and Eloquent Builder classes

### PR DESCRIPTION
**Pull request by rafis**
#closes 19810

### CODE REVIEW

**Comment by tillkruss** [comment]
That kinda reminds me of [this](https://twitter.com/legacy_coder/status/880138870460997632).

**Comment by vlakoff** [comment]
Just in case, I'll stress it out: legacy_coder is going to promote shitty code advices. Don't apply them!

File: src/Illuminate/Database/Concerns/BuildsQueries.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
      * @param  callable  $callback
      * @return bool
```
Comment (lucasmichot) [inline src/Illuminate/Database/Concerns/BuildsQueries.php]: Missing `@param bool $forDelete`

**Comment by taylorotwell** [comment]
I don't really want to add a boolean flag for this. Also it needs to be explained somewhere what the actual problem is other than "it doesnt work".

**Pull request closed without merging**